libcontainer/configs/mount_linux.go

```diff
@@ -0,0 +1,36 @@
+package configs
+
+import "golang.org/x/sys/unix"
+
+type Mount struct {
```

Comment:
IMHO The struct duplication seems not ideal. I see they are mostly identical, except the non linux doesn't have the unix bits.

I'm not sure we need two structs, but even if we do, can't we have one with the common parts and just embed it here?  I mean embed as shown in these examples: https://gobyexample.com/struct-embedding

Reply:
> It would be great if you can paste the compilation error you see (not all the lines, just the important ones if it is too verbose), so we can easily know how this fails and why this fixes it.
>

Updated the PR description 
 

> Also, why do you need this to compile on Mac? I mean, none of the cgroup stuff works there, right? Is it even a supported platform for runc?
> 

Libcontainer configs is utilized in other projects, including Kata Containers, which is adding support for building and running on Darwin.

> I'm fine with the PR, but if we don't want to regress, we should probably add a test to compile on Mac, that I'm not sure what maintainers think about that.

Sure - you don't need to be on Mac. Just run "GOOS=darwin go build".

Reply:
Agreed that it's sub-optimal.

Reply:
I created a base struct which can be used to help dedup, PTAL -- thank for the suggestion!